app/src/main/java/com/lastaoutdoor/lasta/ui/screen/social/components/MessageList.kt

```diff
@@ -66,8 +67,17 @@ fun MessageList(
       }
       // list of messages
       LazyColumn {
-        items(messages.size) {
-          MessageCard(messages[it], friends[it].userId, navigateToConversation)
+        items(messages.size) { messageIndex ->
+          val friendId =
+              messages[messageIndex]
+                  .members
+                  .firstOrNull { friends.map { it.userId }.contains(it.userId) }
+                  ?.userId
+          if (friendId == null) {
+            Log.e("MessageList", "Friend not found in friends list")

```

Comment:
Don't forget to remove the debug console prints

Reply:
It's not really a print, I just wanted to log an error (you can see it as non-blocking throw)